frontend/packages/operator-lifecycle-manager/src/components/operand/create-operand.tsx

```diff
@@ -121,110 +115,88 @@ export const CreateOperand: React.FC<CreateOperandProps> = ({
+        prune={pruneFunc}
+        YAMLEditor={OperandYAML}
+      />
+    </>
   );
 };
 
-const stateToProps = (state: RootState, props: Omit<CreateOperandPageProps, 'model'>) => ({
-  model: state.k8s.getIn(['RESOURCES', 'models', props.match.params.plural]) as K8sKind,
-});
-
-export const CreateOperandPage = connect(stateToProps)((props: CreateOperandPageProps) => {
+const CreateOperandPage: React.FC<CreateOperandPageProps> = ({ match }) => {
   const { t } = useTranslation();
+  const [model] = useK8sModel(match.params.plural);
+  const [csv, loaded, loadError] = useK8sWatchResource<ClusterServiceVersionKind>({
+    kind: referenceForModel(ClusterServiceVersionModel),
+    name: match.params.csvName,
+    namespace: match.params.ns,
+    isList: false,
+  });
+
+  const DefaultPage = React.useCallback(
+    (props: { match: CreateOperandProps['match'] }) => (
+      <StatusBox loaded={loaded} loadError={loadError} data={csv}>
+        <CreateOperand {...props} csv={csv} initialEditorType={EditorType.Form} />
+      </StatusBox>
+    ),
+    [csv, loadError, loaded],
+  );
   return (
     <>
       <Helmet>
-        <title>
-          {t('olm~Create {{item}}', { item: kindForReference(props.match.params.plural) })}
-        </title>
+        <title>{t('olm~Create {{item}}', { item: kindForReference(match.params.plural) })}</title>
       </Helmet>
-      {props.model && (
-        <Firehose
-          resources={[
-            {
-              kind: referenceForModel(ClusterServiceVersionModel),
-              name: props.match.params.appName,
-              namespace: props.match.params.ns,
-              isList: false,
-              prop: 'clusterServiceVersion',
-            },
-            {
-              kind: CustomResourceDefinitionModel.kind,
-              isList: false,
-              name: nameForModel(props.model),
-              prop: 'customResourceDefinition',
-              optional: true,
-            },
-          ]}
-        >
-          {/* FIXME(alecmerdler): Hack because `Firehose` injects props without TypeScript knowing about it */}
-          <CreateOperand
-            {...(props as any)}
-            model={props.model}
-            match={props.match}
-            initialEditorType={EditorType.Form}
+      {loaded && (
+        <div className="co-create-operand__breadcrumbs">
+          <BreadCrumbs
+            breadcrumbs={[
+              {
+                name: csv.spec.displayName,
+                path: resourcePathFromModel(
+                  ClusterServiceVersionModel,
+                  csv.metadata.name,
+                  csv.metadata.namespace,
+                ),
+              },
+              {
+                name: t('olm~Create {{item}}', { item: model.label }),
+                path: window.location.pathname,
+              },
+            ]}
           />
-        </Firehose>
+        </div>
       )}
+      <CreateResource match={match} DefaultPage={DefaultPage} />
```

Comment:
Passing a component as a prop is also an antipattern. We should just load the extensions in this component and conditionally render here.